[#12] 사용자가 등록한 언어 삭제 기능 구현

msugo1 opened the pull request:
## Controller 계층 관련

### LanguageController (new)

- `deleteLanguages` 메소드 추가

: 사용자의 언어 삭제 요청을 전달 받을 메소드

## Service 계층 관련

### LanguageService (new)

- `deleteLanguages` 메소드 추가

1) Controller 가 받은 `삭제 요청`을 전달 받아서 처리하기 위한 메소드

2)  최대로 등록 가능한 언어 수를 초과한 삭제요청이 들어올 시 예외 발생

## Repository/Mapper 계층 관련

### LanguageMapper (new)

- `deleteLanguages` 메소드 추가

: 서비스 계층에서 전달 받은 언어 데이터를 가지고 DB에서 삭제를 처리할
메소드

### LanguageMapper.xml (new)

- 위의 로직에 맞는 `Delete 쿼리` 추가

## Test 관련

- 언어 삭제 관련 로직을 테스트하기 위한 서비스 계층 통합 테스트 작성

1) 최대 추가 가능한 언어 수 제한(16개)를 넘지 않는 경우 삭제 처리가
   이루어 지는지 테스트

2) 언어 ID 뿐만 아니라 해당 사용자의 아이디가 일치하는 언어만 삭제가 이루어지는지 테스트

3) 최대 추가 가능한 언어 수의 제한을 넘는 삭제 요청이 들어오면 예외가
   제대로 발생하는지 테스트

- 포스트맨을 활용해 전체적으로 테스트

Commit:
[#12] 사용자가 등록한 언어 삭제 기능 구현

Controller 계층 관련

* LanguageController (new)

- deleteLanguages 메소드 추가

사용자의 언어 삭제 요청을 전달 받을 메소드

Service 계층 관련

* LanguageService (new)

- deleteLanguages 메소드 추가

Controller 가 받은 삭제 요청을 전달 받아서 처리하기 위한 메소드

최대로 등록 가능한 언어 수를 초과한 삭제요청이 들어올 시 예외 발생

Repository/Mapper 계층 관련

* LanguageMapper (new)

- deleteLanguages 메소드 추가

서비스 계층에서 전달 받은 언어 데이터를 가지고 DB에서 삭제를 처리할
메소드

* LanguageMapper.xml (new)

- 위의 로직에 맞는 `Delete 쿼리` 추가

Test 관련

- 언어 삭제 관련 로직을 테스트하기 위한 서비스 계층 통합 테스트 작성

1) 최대 추가 가능한 언어 수 제한(16개)를 넘지 않는 경우 삭제 처리가
   이루어 지는지 테스트

2) 언어 ID 뿐만 아니라 해당 사용자의 아이디가 일치하는 언어만 삭제가 이루어지는지 테스트

3) 최대 추가 가능한 언어 수의 제한을 넘는 삭제 요청이 들어오면 예외가
   제대로 발생하는지 테스트

- 포스트맨을 활용해 전체적으로 테스트

f-lab-dev commented on src/main/java/me/soo/helloworld/controller/LanguageController.java:
```diff
@@ -24,4 +26,11 @@ public void modifyLanguageLevel(@CurrentUser String userId, @RequestBody Languag
         languageService.modifyLevel(userId, languageDataWrapper.getDataList(), languageDataWrapper.getStatus());
     }
+
+    @DeleteMapping
```
이것도 표현의 방식이 애매할 것 같은데 우리는 `언어`를 삭제하는게 아닌 `유저의 언어`를 삭제하는겁니다.

그러면 **내가 구사할 수 있는 언어**를 표현하려면 `/my-infos/languages` 정도로 표현해주는게 좋지 않을까요?

msugo1 replied:
지금까지 놓치고 있던 부분이네요; 결국 언어를 추가하는 것은 `내가 구사할 수 있는/혹은 배우는 언어`가 맞으니 

`/my-infos/languages` 가 더 적절해 보입니다. 피드백 반영하였습니다. 감사합니다.

Commit:
Merge branch 'feature/11' into feature/12

Commit:
[#12] 피드백 반영

- `LanguageController`의`URL` 변경 `/languages -> /my-infos/languages`
: 결국 언어를 추가/변경/삭제한다는 것은 현재 로그인 중인 유저의 언어
정보를 수정하는 것이므로 URL이 좀 더 명확한 의미를 가지도록 변경

- `modifyLanguge -> modifyLanguages`로 메소드 이름 수정
: 다수의 언어 레벨을 한 번에 수정가능하므로 복수형의 `-s`를 붙임

Commit:
[#12] LanguageController 메소드 파라미터 배치 통일
